nvm.sh

```diff
@@ -2141,13 +2128,11 @@ nvm_check_file_permissions() {
     setopt nonomatch
   fi
   for FILE in $1/* $1/.[!.]* $1/..?* ; do
-      if [ -d "$FILE" ]; then
-        if ! nvm_check_file_permissions "$FILE"; then
-          if [ "${ZSH_HAS_NONOMATCH_UNSET}" -eq 1 ] && nvm_has "setopt"; then
-            setopt nomatch
-          fi
-          return 2
+      if [ -d "$FILE" ] && ! nvm_check_file_permissions "$FILE"; then
```

Comment:
aha, sorry, you're right, there's no else.

However, in the original code, if the file is a directory, nothing happened outside of the `nvm_check_file_permissions` branch - in your new code, if the file is a directory but the permissions check fails, it falls to the `elif` - which checks that it exists, and is not writable and not owned (possible for a directory), which means the file-related error message would be printed.

Am I reading this incorrectly?